File: 338.counting-bits.java
/*
 * @lc app=leetcode id=338 lang=java
 *
 * [338] Counting Bits
 */

// @lc code=start
class Solution {
    public int[] countBits(int n) {
        int[] dp = new int[n + 1];
        for (int i = 1; i <= n; i ++) {
            dp[i] = dp[i >> 1] + (i & 1);
        }
        return dp;
    }
}
// @lc code=end

File: 338.counting-bits.py
#
# @lc app=leetcode id=338 lang=python3
#
# [338] Counting Bits
#

# @lc code=start
from typing import List

class Solution:
    def countBits(self, n: int) -> List[int]:
        output = []
        while n >= 0:
            output.append(bin(n)[2:].count('1'))
            n -= 1
        return output[::-1]
# @lc code=end

